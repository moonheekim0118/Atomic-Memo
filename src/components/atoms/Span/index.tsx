import React from 'react';
import styled from 'styled-components';

interface Props {
  children: React.ReactChild;
  color?: string;
  textAlign?: 'left' | 'right' | 'center';
  width?: string;
  padding?: string;
  size?: 'small' | 'normal' | 'big' | 'title';
}

const Span = ({
  children,
  color,
  textAlign = 'left',
  width = 'auto',
  padding,
  size = 'normal',
}: Props) => {
  const stylingProps = {
    color,
    textAlign,
    width,
    padding,
  };

  return (
    <StyledSpan {...stylingProps} className={size}>
      {children}
    </StyledSpan>
  );
};

const StyledSpan = styled.span<Props>`
  color: ${(props: Props) => props.color || 'black'};
  text-align: ${(props: Props) => props.textAlign};
  width: ${(props: Props) => props.width};
  padding: ${(props: Props) => props.padding};
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;

  &.small {
    font-size: 1rem;
  }

  &.normal {
    font-size: 1.2rem;
  }

  &.big {
    font-size: 1.2rem;
  }

  &.title {
    font-size: 2.5rem;
    font-weight: bold;
  }
`;

export default Span;
