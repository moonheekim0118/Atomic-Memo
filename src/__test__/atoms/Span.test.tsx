import 'jsdom-global/register';
import React from 'react';
import Span from '../../components/atoms/Span';
import { mount } from 'enzyme';

describe('<Span/>', () => {
  let container = null;

  it('renders correctly', () => {
    container = mount(<Span size="big">test</Span>);
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
