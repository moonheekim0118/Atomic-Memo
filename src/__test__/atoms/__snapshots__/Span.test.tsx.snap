// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Span/> matches snpashot 1`] = `"<span width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG big\\">test</span>"`;
